### ISSUE TITLE
changelog v126 [important: read upcoming changes for FF128]

### ISSUE THREAD

**Issue by Thorin-Oakenpants** [arkenfox/user.js#1846]
## :green_square: v126

[FF126 release notes](https://www.mozilla.org/firefox/126.0/releasenotes/)
[FF126 for developers](https://developer.mozilla.org/docs/Mozilla/Firefox/Releases/126)
[FF126 security advisories](https://www.mozilla.org/security/advisories/mfsa2024-21/)

---

## ⭐ ⚠️ there is a migration of prefs coming in FF128 for sanitizing (on close and manually), including new ones,

so make sure to add any new corresponding sanitizing prefs to your overrides if required before 128

## ⭐ ⚠️ in FF128 I will also move arkenfox to using FPP not RFP see #1804 

if you want to continue to use RFP (4501) and/or LB (4504) and/or disable webgl (4520) then you might as well add them to your overrides as well, so I don't change them on you without warning.

see [this comment below](https://github.com/arkenfox/user.js/issues/1846#issuecomment-2153830049) for my overrides

---

### CHANGELOG

- [all changes](https://github.com/arkenfox/user.js/compare/122.0...126.0)
- ToDo: diffs
   - [FF122-FF123](https://github.com/arkenfox/user.js/issues/1814)
   - [FF123-FF124](https://github.com/arkenfox/user.js/issues/1820)
   - [FF124-FF125.0.1](https://github.com/arkenfox/user.js/issues/1836)
   - [FF125.0.1-FF126](https://github.com/arkenfox/user.js/issues/1842)
- v126 PR is at #1816

---

- new and active in user.js 126
  - for spoof english see #1827
```js
user_pref("browser.contentanalysis.default_allow", false); // [FF124+] [DEFAULT: false]
user_pref("browser.urlbar.yelp.featureGate", false); // [FF124+] [DEFAULT: false]
user_pref("privacy.spoof_english", 1);
```

- new and inactive in user.js 126
   - for GPC see #1818
```js
   // user_pref("browser.link.force_default_user_context_id_for_external_opens", true);
   // user_pref("browser.urlbar.quicksuggest.enabled", false); // [FF92+] [DEFAULT: false]
   // user_pref("privacy.fingerprintingProtection.remoteOverrides.enabled", false); // [FF127+]
   // user_pref("privacy.globalprivacycontrol.enabled", true);
```

- new in user.js 126, required for 128
  - see #1837
  - `clearSiteData` FF128+ = `Privacy & Security>Browser Privacy>Cookies and Site Data>Clear Data`
  - `privacy.cpd` old prefs migrate to `clearHistory`
  - `privacy.clearOnShutdown` migrates to `clearOnShutdown_v2`
  - migration (the prefs are reduced) is documented at [here](https://github.com/arkenfox/user.js/issues/1837#issuecomment-2142704385)
  - I have kept the same values as before, so all you need to do is update your overrides to suit
```js
user_pref("privacy.clearHistory.cache", true);
user_pref("privacy.clearHistory.historyFormDataAndDownloads", true);
user_pref("privacy.clearHistory.cookiesAndStorage", false);
   // user_pref("privacy.clearHistory.siteSettings", false);

user_pref("privacy.clearOnShutdown_v2.cache", true);  // [FF128+] [DEFAULT: true]
user_pref("privacy.clearOnShutdown_v2.historyFormDataAndDownloads", true); // [FF128+] [DEFAULT: true]
   // user_pref("privacy.clearOnShutdown_v2.siteSettings", false); // [FF128+] [DEFAULT: false]
user_pref("privacy.clearOnShutdown_v2.cookiesAndStorage", true) // Cookies, Site Data, Active Logins [FF128+]

user_pref("privacy.clearSiteData.cache", true);
user_pref("privacy.clearSiteData.cookiesAndStorage", false); // keep false until it respects "allow" site exceptions
user_pref("privacy.clearSiteData.historyFormDataAndDownloads", true);
   // user_pref("privacy.clearSiteData.siteSettings", false);
```

- made inactive in user.js 126
   - they are default false
```
   // user_pref("browser.urlbar.suggest.quicksuggest.nonsponsored", false); // [FF95+] [DEFAULT: false]
   // user_pref("browser.urlbar.suggest.quicksuggest.sponsored", false); // [FF92+] [DEFAULT: false]
```

- moved to `9999: DEPRECATED / REMOVED`
```js
user_pref("browser.messaging-system.whatsNewPanel.enabled", false); // deprecated FF126
user_pref("browser.ping-centre.telemetry", false); // deprecated FF123
```

---

STATS

```
 STATS v126: up to and including section 4500, minus the parrots
 =========
    total: 192
 inactive:  50
      n/a    9 (FF128+: clearHistory, clearOnShutdown_v2, clearSiteData prefs)
           ---
   active: 133
  default:  23 (at least)
      n/a:   2 (of the three prefs in 0204, only one will apply)
           ---
  flipped: 108 (at most)
```

**Comment by Thorin-Oakenpants** [comment]
![allhail-AF](https://github.com/arkenfox/user.js/assets/16656956/e8bab124-c3d3-49f1-8c8a-f36b10ce0d15)

**Comment by Thorin-Oakenpants** [comment]
all up, very boring .. **only 1 new active pref** (spoof english) which if anyone is already using it, it should be in their overrides. 

Everything else is upcoming (FF128 sanitizing migration), or commented out since it's at default-what-we-want (or for prefsCleaner), or deprecated

Enjoy the stability of arkenfox :)

**Comment by Thorin-Oakenpants** [comment]
https://github.com/arkenfox/user.js/pull/1847/commits/10fddc8072b69cb7f3448fb4c5e09c8bc23b72f3

my overrides for FF128 - add em now if you want to continue with RFP in FF128+
```js
user_pref("privacy.resistFingerprinting", true);
user_pref("privacy.resistFingerprinting.letterboxing", true);
user_pref("webgl.disabled", true);
user_pref("privacy.spoof_english", 2); // required, I have en-US app lang but my locale !== en-US
```

FYI: re spoof english and en-US on english but not en-US OSes
- https://bugzilla.mozilla.org/show_bug.cgi?id=1739712#c5
- > Soo, we devised this slightly sophisticated system where if the language portion of your OS locale matches langauge portion of Firefox locale (for example "en-US" and "en-GB"), then we will use the region portfion of your OS locale in Firefox.

**Comment by Thorin-Oakenpants** [comment]
no, newwin (max sizes) is only used when RFP is enabled

**Comment by Thorin-Oakenpants** [comment]
I'll add some info to #1804 tomorrow hopefully and unlock the topic and everyone can go have a good yarn and discuss it to death :) I of course will unsubscribe having said my bit :) e.g. why I plan to keep using RFP

**Comment by eleius** [comment]
Noob question: I couldn't find `browser.search.serpEventTelemetryCategorization.enabled` in the new user.js, should I disable it (as suggested [here](https://www.ghacks.net/2024/05/14/firefox-126-telemetry-privacy-feature-and-security-fixes/)) or is there a master switch somewhere already?

**Comment by rusty-snake** [comment]
#1840 ?

**Comment by Thorin-Oakenpants** [comment]
@Tiagoquix IDK - languages is just languages - i.e request page in x,y,z ... and the app language can be different if you want

So you have en-US interface, `en-US,en` languages - that's all groovey (app language is used by webcontent for a lot of things, e.g. look and feel for widgets: buttons, input fields, etc - which to me is backwards but probably a legacy thing: they should eventually make all chrome use app lang, and all web content use web-request lang, but that also has issues)

Anyway, so what really matters here is the `locale` - all the Intl....resolvedOptions. If you have `en-US` and your os is en-US you're cool, but if your os is en-CA or en-GB then it uses that. Same goes for the locales for spanish, french, german etc. But if you have `en-US` and your os is `es-ES` then it will use `en-US` (or so I believe, I do not have a non-english OS handy)

And that's before you start using spoof_english, which only allows spoofing for a single language, not all (because of all the places the app language leaks or is used in web content)

And resetting spoof english also has issues and I think bad design - it does not reset things which actually caused a major regression in Tor Browser where users could mix and match languages and locales - e.g. german language and english locale - relaly bad.

All I can say is test it: https://arkenfox.github.io/TZP/tzp.html#region

**Comment by Thorin-Oakenpants** [comment]
I already updated the live user.js with the active missing `;` - updater will pick up on this. The other is commented out and non-breaking IMO, and is a FF127+ pref recommended not to use (but I get that this is about the syntax part)

What am I missing here? Do I really need to a 126.1 release?

**Comment by Thorin-Oakenpants** [comment]
https://github.com/arkenfox/user.js/compare/126.0...126.1

**Comment by Thorin-Oakenpants** [comment]
`https://github.com/Floorp-Projects/Floorp/discussions/1122` - LOLs .. they haven't got a clue what privacy is, and they actively undermine it with dumb decisions - I hope they reject the proposal and do not tarnish arkenfox's sterling record

**Comment by Thorin-Oakenpants** [comment]
now at [31 spartas](https://www.imdb.com/title/tt0416449/)

![31 spartas](https://github.com/arkenfox/user.js/assets/16656956/439cf608-622e-4592-ba12-bbcbf7aac936)

[last time I counted in sparta units](https://github.com/arkenfox/user.js/issues/1277#issuecomment-977544351) cc: @bagder I will catch you :)

**Comment by MagicalDrizzle** [comment]
may I ask if the `cpd` migration to `clearHistory` already happened? because there are only the latter in user.js, as opposed to the v2 prefs where we have both until 128 rolls out

**Comment by Thorin-Oakenpants** [comment]
> because there are only the latter in user.js

they are both in the user.js - migration should not be happening until 128 AFAIK

https://github.com/arkenfox/user.js/blob/47cbf5b9740ef59ed866874346d3fee3379f8da3/user.js#L691-L710

**Comment by Thorin-Oakenpants** [comment]
without my support,. users can also use FPP (default) but kick in RFPTargets - so all RFP minus the bits that break

**Comment by Janaue** [comment]
yeah the topic and the acronyms are already hard to follow, and then the changes by mozilla year after year makes it more nebulous lol

**Comment by Thorin-Oakenpants** [comment]
https://github.com/arkenfox/user.js/blob/6446d73cf572fcdf631534a5a51276a64eec4a2d/user.js#L799

for those wondering why
- almost no-one, if anyone, has UTC as a system/browser timezone, even if it is supported
- for this reason, browsers with UTC are associated with Tor Browser and are penalized
- since we only really care about recent (and future) date-times, we wanted to change to a real world timezone used by actual people
- 16 timezone names met the criteria that they use the same timezone (as UTC) and do not change with DST (daylight savings time)
- 15 were in Africa .. and one was `Atlantic/Reykjavik` - we chose Iceland
- since 1916, all date-times in this timezone are effectively the same as UTC/GMT, it's just a different name
- now if a script wants to penalize Tor Browser (RFP, Mullvad Browser) based on timezone name, then they can also upset all of Iceland, and those fuckers are descended from Vikings
- https://gitlab.torproject.org/tpo/applications/tor-browser/-/issues/42397
   - solution instigated by some genius

**Comment by Thorin-Oakenpants** [comment]
> since we only really care about recent (and future) date-times

to clarify, that means usability for the user with their calendars, current day/date/time etc = to stay consistent all year round with UTC. Historically dates differ (prior to 1916 I from memory - but all users are the same, so not an entropy or usability issue)

**Comment by GlitteringReturn** [comment]
Holly crap, I remember barely reading through the Wiki, implementing Arkenfox into my browser, albeit not understanding the update & maintain section. A couple months of University went by, I read this, and I have no clue what is going on tbh. Anyway, two questions, (1) can we cut out all the BS so to say and make a concise summary and guide to implement and maintain Arkenfox, only with integral steps and information necessary to get it running, in one central location (i.e., 1 webpage or document), this would really help put people who do not have a background in computer science (2) this is kinda pointless, but can we add rounded edges to the letterbox like Mullvad?

**Comment by hankertrix** [comment]
https://github.com/arkenfox/user.js/issues/1080#issue-774750296

The override here needs to be updated with the migrated prefs for history, specifically:

```js
user_pref("privacy.clearOnShutdown.history", false);
```

To:

```js
user_pref("privacy.clearOnShutdown_v2.historyFormDataAndDownloads", true);
```

**Comment by Thorin-Oakenpants** [comment]
^DONE

**Comment by Thorin-Oakenpants** [comment]
https://github.com/arkenfox/user.js/issues/1846#issuecomment-2183064738

LMFAO .. once again, do not use entropy figures from fucking test sites, they are **BULLSHIT**

![bollocks](https://github.com/user-attachments/assets/31cfcb3b-d3df-404b-ba49-ca193c75774d)

So it's been less than a week or so since RFP in FF128+ took effect (and a bit longer in TB/MB stable). If I remember, I will revisit in a couple of months time, where I expect that number to drop significantly (not that it matters but it is hilarious)

So apparently 1 in 23 users are on Icelandic timezone. Whilst not all countries have equal internet access or usage, and population is not a perfect measure .. let's just examine how ludicrous this estimation is

back of the napkin math
- the population of Iceland is 400k
- all things being equal, if Iceland is 1/23rd this would make the world's population 9.2mn
   - the world's population is ~ 8.1 billion .. we'll call it 8 billion
   - that's off by a factor of 870 .. i.e 870 x 9.2mn = 8bn
- lets say that only half the world has internet access but ALL of Iceland does, that's still skewed by a factor of 435

**Comment by Thorin-Oakenpants** [comment]
I summarized it for you

> once again, do not use entropy figures from fucking test sites, they are BULLSHIT

**Comment by Thorin-Oakenpants** [comment]
out of interest I thought I would actually list the amiunique tests and how relevant they are for TB entropy wise. This "protection" rather than "entropy" applies to RFP and arkenfox as well except for
- fonts: RFP uses a different mechanism but is just as tight on windows and mac, but linux is very hit and miss IIUIC (I need to check it since I last did a year ago now FPP is here), whereas TB bundles most or all fonts (desktop) and covers way more script support, and linux is super tight (no system fonts used, all bundled)
- webGL: we disable it but TB allows some of it in standard but nixes readPixels. Both are not great options.

So the thing with TB/MB is that they are a crowd. And how you defeat FPing is to take metric by metric and make it useless (hide the real value and reduce to all the same or equivalency). We do not care about other browsers, or BS entropy figures, when we have math and science (and common sense) on our side

immediately 32% of metrics do not even apply and another 39% are ineffective (we know because we enforce a value, or we set defaults - we should harden some of them, and plan to) and another 10% are equivalency. The header items not included in this list below are equivalency of some JS items.

```
19/59 non-gecko/pointless
---
17 - Java enabled
18 - Device memory
43 - Accelerometer
44 - Gyroscope
45 - Proximity sensor
46 - Keyboard layout
47 - Battery
48 - Connection
49 - key
50 - Location bar
51 - Menu bar
52 - Personal bar
53 - Status bar
54 - Tool bar
55 - Result state
56 - List of fonts (Flash)
57 - Screen resolution (Flash)
58 - Language (Flash)
59 - Platform (Flash) 

23/59 all users the s/be the same (unless users add extensions/meddle with prefs)
---
3 - Cookies enabled
4 - Timezone
6 - Canvas
8 - Use of Adblock
9 - Do Not Track
10 - Navigator properties
11 - BuildID
12 - Product
13 - Product sub
14 - Vendor
15 - Vendor sub
16 - Hardware concurrency
19 - List of plugins
22 - Screen depth
23 - Screen available top
24 - Screen available Left
27 - Screen left
28 - Screen top
29 - Permissions
34 - Use of local storage
35 - Use of session storage
36 - Use of IndexedDB
42 - Media devices

6/59 equivalency of OS or language (or for audio x86/amd vs ARM when enabled, e.g. in MB14)
---
1 - User agent
2 - Platform
5 - Content language
38 - Audio context
39 - Frequency analyser
40 - Audio data

~~11~~ 4 things left
---
7 - List of fonts
   - desktop mostly mitigated by bundling and allowlisting
20-26 - Screen width height availableHeight availableWidth
   - all screen/inner largely mitigated with newwin & letterboxing
30-33 - WebGL Vendor Renderer Data Parameters
   - webgl needs work
37+41 - Audio/Video formats
   - formats needs work
```

tl;dr: TB is world leading and can deal with unsophisticated tests like amiunique

**Comment by Thorin-Oakenpants** [comment]
AF defaults (minus the old prefs which shouldn't be used if you have migrated and commented out items)
```js
// shutdown
user_pref("privacy.clearOnShutdown_v2.cache", true);
user_pref("privacy.clearOnShutdown_v2.historyFormDataAndDownloads", true);
user_pref("privacy.clearOnShutdown_v2.cookiesAndStorage", true);

// manual clear data: Privacy & Security>Browser Privacy>Cookies and Site Data>Clear Data
user_pref("privacy.clearSiteData.cache", true);
user_pref("privacy.clearSiteData.cookiesAndStorage", false);
user_pref("privacy.clearSiteData.historyFormDataAndDownloads", true);

// manual clear history: Ctrl-Shift-Del | Privacy & Security>History>Custom Settings>Clear History
user_pref("privacy.clearHistory.cache", true);
user_pref("privacy.clearHistory.historyFormDataAndDownloads", true);
user_pref("privacy.clearHistory.cookiesAndStorage", false);
```

what you added as overrides: AFAICT this should work
```js
// shutdown
user_pref("privacy.clearOnShutdown_v2.historyFormDataAndDownloads", false);
user_pref("privacy.clearOnShutdown_v2.cookiesAndStorage", false);

// manual clear data
user_pref("privacy.clearSiteData.cache", false);
user_pref("privacy.clearSiteData.historyFormDataAndDownloads", false);

// manual clear history
user_pref("privacy.clearHistory.cache", false);
user_pref("privacy.clearHistory.historyFormDataAndDownloads", false);
```

now tell me WHERE in the three sections, with your overrides, are you seeing `clear history` checked? One of them, all of them? What does your overrides parrot say in about:config? There may be a bug upstream

**Comment by Thorin-Oakenpants** [comment]
> Isn't he just talking about "Clear history when Firefox closes" in the settings? Which is privacy.sanitize.sanitizeOnShutdown

yup - but I want to know what happens in all three

**Comment by AeliusSaionji** [comment]
I got bit by this.

The v2 historyformdata also controls history. You can see the setting box and about:config update each other in real time.

`privacy.clearOnShutdown_v2.historyFormDataAndDownloads`


![image](https://github.com/user-attachments/assets/223dcc56-7bf4-47c2-adff-cde564fd4e8d)
![image](https://github.com/user-attachments/assets/1a637824-c93e-4241-8c0d-73b7accd6590)

**Comment by rusty-snake** [comment]
> The v2 ***history***formdata also controls history.

**Comment by AeliusSaionji** [comment]
Indeed, but with the new change to combine two settings into one key combined with not being intimately familiar with all the available options has created something of a nasty trap.

I freely admit that I did not realize until your post that the history is its own entity in that key, rather than referring to a history of form data.

**Comment by Thorin-Oakenpants** [comment]
> rather than referring to a history of form data

so I listed the new item under the old items - so cache -> v2.cache etc

https://github.com/arkenfox/user.js/blob/ff5c959cb9b5a65ede9aab82b00cbb283b160e14/user.js#L653-L658

and here I even listed the combined items in // comments

https://github.com/arkenfox/user.js/blob/ff5c959cb9b5a65ede9aab82b00cbb283b160e14/user.js#L677-L680

It also makes no sense to call something  a `history of form data` and `downloads` or a history of `form data and downloads` - when sanitizing form data, downloads you are wiping them, the word history is redundant. The word history refers to history as it's own entity

Anyway ... I have to admit it's all a little messy, roll on EOL of ESR115 and I can remove all the old prefs

**Comment by AeliusSaionji** [comment]
> makes no sense

You're right. Now that this has been brought to my full attention and scrutiny, I would find it silly and redundant were it named such.

However, I am only human, and at the time I was reviewing the settings, the dots took a while to connect- the odd name didn't make me immediately realize the old history setting had no new equivalent.

**Comment by mherrmann3** [comment]
I too got trapped by the naming of `historyFormDataAndDownloads`, so I forgot to override it with `false` in June, believing it doesn't affect history. Btw: it even **clears tabs**!
I was wondering why the FF devs didn't call it `history*And*FormDataAndDownloads` (`*` for emphasis) or `historyFormdataDownloads` (no `and` and small `d` in `Formdata`)...

**Comment by rusty-snake** [comment]
Likely "history, form data and downloads" then remove non-alphanumeric characters (`,`) and switch to camelCase for word splitting. (The name of internal preferences must be understood by its developers, not by endusers).

>  Btw: it also clears tabs!

In Ctrl-Shift-Entf or during session restore? Because during SR this is known as it was with the old system.

**Comment by mherrmann3** [comment]
> In Ctrl-Shift-Entf or during session restore? Because during SR this is known as it was with the old system.

Yes, with the session restore override `("browser.startup.page", 3)` of #1080 and `("privacy.clearOnShutdown_v2.historyFormDataAndDownloads", true)` (when it was not yet in the recipe). Okay, good to know; now I found the corresponding doc 😬:
https://github.com/arkenfox/user.js/blob/ff5c959cb9b5a65ede9aab82b00cbb283b160e14/user.js#L85

**Comment by Thorin-Oakenpants** [comment]
go vikings (i am now responsible for about 7 visits)

![go vikings](https://github.com/user-attachments/assets/b5bcc234-dc03-4b4e-98a2-150c8f1e20ab)

**Comment by Thorin-Oakenpants** [comment]
ok, it's been a month .. let's check in on our viking fiends

![grey goo](https://github.com/user-attachments/assets/d81d404e-3622-4b15-b8db-9ab605f1ffd3)

**Comment by ntnguyen1234** [comment]
Hi, when I checked https://arkenfox.github.io/TZP/tzp.html with arkenfox' default user.js and new Firefox 129 profile, I see 2 values related to `EXSLT` that are marked red: 1 in `[offset] timezone` section which shows my timezone and the other in `timing precision`.

I found the timezone has been patched in 131: https://bugzilla.mozilla.org/show_bug.cgi?id=1891690

However I couldn't find much information the other. Anyone has more information about this? Is it addressed together with the timezone issue in 131? Thank you.

**Comment by Thorin-Oakenpants** [comment]
links
- https://bugzilla.mozilla.org/show_bug.cgi?id=1912129
- https://gitlab.torproject.org/tpo/applications/tor-browser/-/issues/30862

so back in the day timing was set at 100ms but in FF102 this was changed to link to rAF (animation frames) and set at 60 FPS == 16.6667ms which solved a lot of issues (but also still causes issues). Not really all that productive _now_ to limit exslt timing from 10ms (there are easy ways to get 1ms timing) since we're already at 16.7ms but why not - 1sec it is.

Both exslt timezone and timing will be backported to ESR for Tor Browser

**Comment by ntnguyen1234** [comment]
Thank you for your information 🙂

**Comment by Thorin-Oakenpants** [comment]
jfc ...

![JFC](https://github.com/user-attachments/assets/f67080d0-fd53-4c42-9a87-09e995d7ce3a)
